### ISSUE TITLE
Please provide an example how to get and delete a connection

### ISSUE THREAD

**Issue by pokecheater** [retejs/rete#721]
### Description

Hi, 

I stumbled with the update from vue2 to vue3 over the following problem: When I update my outputsockets of an already defined and connected node (since the underlying datamodel has changed) I had to remove the old sockets first. 

When a socket is already connected this leads to broken left over connections that point into nirvana:

<img width="305" alt="image" src="https://github.com/user-attachments/assets/80864f2f-df5e-4da9-bb3f-3c021290488b" />

My workaround in your old version was to indentify those connections right at the socket and to call my remove Connection routine:

<img width="441" alt="image" src="https://github.com/user-attachments/assets/ac569892-61f8-4147-9c36-88c5d7c47c02" />

but now the connections are not available at the socket itself anymore. So how to identify my connections that belong to a certain socket and how to delete them afterwards (please note: that my nodes have a reference to the editor as I patched them already).

Greetz and thx in advance

### Suggested Solution

Provide an example for advanced connection handling.

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by pokecheater** [comment]
For me it would be also fine, if you can tell me how to remove a certain connection instead of fixing it in your code. I already builded a workaround for the same issue in your old vue2 version. But with the new version I have no idea how to delete the connection. If you can please tell me how to do, I can build my own workaround againg.

**Comment by Ni55aN** [comment]
`removeConnection` accepts a connection id, not a connection instance. This is not mentioned in the guides, but can be found in the API docs https://retejs.org/docs/api/rete#nodeeditor-removeconnection

**Comment by Ni55aN** [comment]
Additionally, since this method is asynchronous, it must have an await so that subsequent operations (such as removing the output) are guaranteed to perform after the connection is removed.

**Comment by pokecheater** [comment]
Yep, thx a lot. I solved by defining an additonal function at my base node class that handles the delete process:

```javascript
    async deleteConnections() {
        const connections = this.getConnections();
        for (const connection of connections) {
            this.editor.removeConnection(connection.id);
        }
    }
```

And in my ReteJSEditor Setup class/Module

I can now rely on that function

```javascript
 deleteSelectedNodes() {
    const selected_nodes = this.getSelectedNodes();
    for (const node of selected_nodes) {
      node.deleteConnections();
      this.editor.removeNode(node.id);
    }
  }
```
